pkg/util/workloadspread/workloadspread.go

```diff
@@ -392,31 +397,204 @@ func (h *Handler) tryToGetTheLatestMatchedWS(matchedWS *appsv1alpha1.WorkloadSpr
 	return wsClone, nil
 }
 
+// handleWorkloadChange handle the update event of workload in case that webhook cannot wait for controller to be
+// done. In this function, if some errors occurred, we will regard pod as the latest, and continue to follow the
+// MissingReplicas(original) logic.
+func (h *Handler) handleWorkloadChange(ws *appsv1alpha1.WorkloadSpread, pod *corev1.Pod, operation Operation) (bool, int32, error) {
+	switch operation {
+	case CreateOperation, EvictionOperation, DeleteOperation:
+	default:
+		return false, -1, nil
+	}
+
+	var workloadReplicas int32 = 0
+	switch ws.Spec.TargetReference.Kind {
+	// Deployment Handler
+	case controllerKindDep.Kind:
+		// If the pod is managed by Deployment, 'pod-template-hash' label should not be empty.
+		podTemplateHash, ok := pod.Labels[appsv1.DefaultDeploymentUniqueLabelKey]
+		if !ok {
+			return false, -1, nil
+		}
+
+		// try to get the deployment from informer cache
+		// TODO: we should consider how to solve informer latency problem
+		deployment := &appsv1.Deployment{}
+		depKey := types.NamespacedName{Name: ws.Spec.TargetReference.Name, Namespace: ws.Namespace}
+		if err := h.Client.Get(context.TODO(), depKey, deployment); client.IgnoreNotFound(err) != nil {
+			return false, -1, err
+		} else if errors.IsNotFound(err) {
+			// In creation case, this pod most likely is the latest revision, because this deployment cache is not synced by our informer, how fresh is this pod!
+			// In deletion/eviction case, if this deployment is not found, the pod most likely is dangling pod, we can ignore this case, because the workload has gone!
+			// Actually, in most cases, this condition will not be true, because we have returned in HandlePodCreation func if deployment is not found.
+			klog.Warningf("WorkloadSpread(%v/%v) cannot find Deployment when handle Pod(%v) create", ws.Namespace, ws.Name, pod.Name)
+		}
+
+		// deployment is not deleted and is cached by this webhook informer
+		if deployment != nil {
+			if deployment.Spec.Replicas != nil {
+				workloadReplicas = *deployment.Spec.Replicas
+			} else {
+				// deployment default replicas is 1
+				workloadReplicas = 1
+			}
+
+			replicaset := h.GetOwnerReplicaSetFor(pod)
+			// if the replicaset is not the latest revision, return false
+			if replicaset != nil && !util.EqualIgnoreHash(&deployment.Spec.Template, &replicaset.Spec.Template) {
```

Comment:
We should check who is the newer one between deployment and replicaset via `Deployment Revision Annotation`.